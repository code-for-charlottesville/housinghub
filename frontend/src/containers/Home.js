import React from "react";
import "../style/App.css";
import { connect } from "react-redux";
import { LoadingSpinner } from "../components/LoadingSpinner";
import { Media } from "react-bootstrap";

class Home extends React.Component {
  componentDidMount() {
    // TODO: fetch things
  }

  render() {
    if (this.props.loading)
      return (
        <div className="container App-header">
          <LoadingSpinner />
        </div>
      );

    return (
      <div>
        <section className="section">
          <div className="container">
            <h1 className="title">{`Welcome, ${this.props.username}`}</h1>
          </div>
        </section>
        <section className="section">
          <div className="container">
            <h1 className="title">My Properties</h1>
            <Media>
              <img
                width={64}
                height={64}
                className="mr-3"
                src="https://cdngeneral.rentcafe.com/dmslivecafe/3/317695/Exterior%20View%20of%20Oxmoor%20Apartments.jpg?crop=(0,20,300,200)&cropxunits=300&cropyunits=200&quality=85&scale=both&"
                alt="Generic placeholder"
              />
              <Media.Body>
                <h5>525 West Main Street</h5>
                <p>
                  Cras sit amet nibh libero, in gravida nulla. Nulla vel metus
                  scelerisque ante sollicitudin commodo. Cras purus odio,
                  vestibulum in vulputate at, tempus viverra turpis. Fusce
                  condimentum nunc ac nisi vulputate fringilla. Donec lacinia
                  congue felis in faucibus.
                </p>
              </Media.Body>
            </Media>{" "}
          </div>
        </section>
        <section className="section">
          <div className="container">
            <h1 className="title">Recent Properties</h1>
            <Media>
              <img
                width={64}
                height={64}
                className="mr-3"
                src="https://www.brac.com/photos/30/xlarge/30_111128_031732_9694.jpg"
                alt="Generic placeholder"
              />
              <Media.Body>
                <h5>5254 Ridge Street</h5>
                <p>
                  Cras sit amet nibh libero, in gravida nulla. Nulla vel metus
                  scelerisque ante sollicitudin commodo. Cras purus odio,
                  vestibulum in vulputate at, tempus viverra turpis. Fusce
                  condimentum nunc ac nisi vulputate fringilla. Donec lacinia
                  congue felis in faucibus.
                </p>
              </Media.Body>
            </Media>
            <Media>
              <img
                width={64}
                height={64}
                className="mr-3"
                src="https://lh3.googleusercontent.com/proxy/OeVKE4Q0tMvsVsbshykH5CGiTicQ-0vqNobrsNhPZwGAOwL5G158dZuzAZaf2oE6O-D6-b4dzFF242rhmXbQa8yXXSVeGDxXS-DBkaOiNBygn_E3FfVAltO4"
                alt="Generic placeholder"
              />
              <Media.Body>
                <h5>99 Avon Street</h5>
                <p>
                  Cras sit amet nibh libero, in gravida nulla. Nulla vel metus
                  scelerisque ante sollicitudin commodo. Cras purus odio,
                  vestibulum in vulputate at, tempus viverra turpis. Fusce
                  condimentum nunc ac nisi vulputate fringilla. Donec lacinia
                  congue felis in faucibus.
                </p>
              </Media.Body>
            </Media>
          </div>
        </section>
      </div>
    );
  }
}

function mapStateToProps(state) {
  return {
    loading: state.appState.loading,
    username: state.user.uid,
  };
}
export default connect(mapStateToProps)(Home);
