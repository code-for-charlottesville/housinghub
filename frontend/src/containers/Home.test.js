import React from "react";
import renderer from "react-test-renderer";
import configureStore from "redux-mock-store";
import sampleMockStore from "../../test/mockStore.js";
import Home from "./Home";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";

const mockStore = configureStore([]);

describe("containers/Home", () => {
  let _matchesSnapshot = (C, props = {}) => {
    let store = mockStore({
      ...sampleMockStore,
      appState: {
        ...sampleMockStore.appState,
        loading: props.loading,
        uid: props.uid,
      },
    });
    const tree = renderer
      .create(
        <MemoryRouter initialEntries={["/my/initial/route"]}>
          <Provider store={store}>
            <C {...props} />
          </Provider>
        </MemoryRouter>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  };

  let testTable = [
    {
      name: "renders spinner if loading",
      component: Home,
      props: {
        loading: true,
        uid: "davidrulz",
      },
    },
    {
      name: "renders full media when not loading",
      component: Home,
      props: {
        loading: false,
        uid: "davidrulz",
      },
    },
  ];

  testTable.forEach((t) => {
    it(t.name, () => {
      _matchesSnapshot(t.component, t.props);
    });
  });
});
