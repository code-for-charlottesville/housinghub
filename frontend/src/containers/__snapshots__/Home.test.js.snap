// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`containers/Home renders full media when not loading 1`] = `
<div>
  <section
    className="section"
  >
    <div
      className="container"
    >
      <h1
        className="title"
      >
        Welcome, 2ad0a95a-8e94-4c2b-b6aa-4c07fb7a1575
      </h1>
    </div>
  </section>
  <section
    className="section"
  >
    <div
      className="container"
    >
      <h1
        className="title"
      >
        My Properties
      </h1>
      <div
        className="media"
      >
        <img
          alt="Generic placeholder"
          className="mr-3"
          height={64}
          src="https://cdngeneral.rentcafe.com/dmslivecafe/3/317695/Exterior%20View%20of%20Oxmoor%20Apartments.jpg?crop=(0,20,300,200)&cropxunits=300&cropyunits=200&quality=85&scale=both&"
          width={64}
        />
        <div
          className="media-body"
        >
          <h5>
            525 West Main Street
          </h5>
          <p>
            Cras sit amet nibh libero, in gravida nulla. Nulla vel metus scelerisque ante sollicitudin commodo. Cras purus odio, vestibulum in vulputate at, tempus viverra turpis. Fusce condimentum nunc ac nisi vulputate fringilla. Donec lacinia congue felis in faucibus.
          </p>
        </div>
      </div>
       
    </div>
  </section>
  <section
    className="section"
  >
    <div
      className="container"
    >
      <h1
        className="title"
      >
        Recent Properties
      </h1>
      <div
        className="media"
      >
        <img
          alt="Generic placeholder"
          className="mr-3"
          height={64}
          src="https://www.brac.com/photos/30/xlarge/30_111128_031732_9694.jpg"
          width={64}
        />
        <div
          className="media-body"
        >
          <h5>
            5254 Ridge Street
          </h5>
          <p>
            Cras sit amet nibh libero, in gravida nulla. Nulla vel metus scelerisque ante sollicitudin commodo. Cras purus odio, vestibulum in vulputate at, tempus viverra turpis. Fusce condimentum nunc ac nisi vulputate fringilla. Donec lacinia congue felis in faucibus.
          </p>
        </div>
      </div>
      <div
        className="media"
      >
        <img
          alt="Generic placeholder"
          className="mr-3"
          height={64}
          src="https://lh3.googleusercontent.com/proxy/OeVKE4Q0tMvsVsbshykH5CGiTicQ-0vqNobrsNhPZwGAOwL5G158dZuzAZaf2oE6O-D6-b4dzFF242rhmXbQa8yXXSVeGDxXS-DBkaOiNBygn_E3FfVAltO4"
          width={64}
        />
        <div
          className="media-body"
        >
          <h5>
            99 Avon Street
          </h5>
          <p>
            Cras sit amet nibh libero, in gravida nulla. Nulla vel metus scelerisque ante sollicitudin commodo. Cras purus odio, vestibulum in vulputate at, tempus viverra turpis. Fusce condimentum nunc ac nisi vulputate fringilla. Donec lacinia congue felis in faucibus.
          </p>
        </div>
      </div>
    </div>
  </section>
</div>
`;

exports[`containers/Home renders spinner if loading 1`] = `
<div
  className="container App-header"
>
  <div
    className="spinner-border"
    role="status"
  />
</div>
`;
